modules/utils/src/merkle.ts

```diff
@@ -1,26 +1,40 @@
+import * as merkle from "@graphprotocol/vector-merkle-tree";
 import { CoreTransferState } from "@connext/vector-types";
 import { HashZero } from "@ethersproject/constants";
 import { keccak256 } from "ethereumjs-util";
 import { MerkleTree } from "merkletreejs";
 
-import { hashCoreTransferState } from "./transfers";
-
-export const generateMerkleTreeData = (transfers: CoreTransferState[]): { root: string; tree: MerkleTree } => {
-  // Sort transfers alphabetically by id
-  const sorted = transfers.sort((a, b) => a.transferId.localeCompare(b.transferId));
+import { encodeCoreTransferState, hashCoreTransferState } from "./transfers";
 
+export const generateMerkleTreeData = (transfers: CoreTransferState[]): { root: string; tree: merkle.Tree } => {
   // Create leaves
-  const leaves = sorted.map((transfer) => {
-    return hashCoreTransferState(transfer);
+  const tree = new merkle.Tree();
+  transfers.forEach((transfer) => {
+    tree.insert_hex_js(encodeCoreTransferState(transfer));
   });
 
-  // Generate tree
-  const tree = new MerkleTree(leaves, keccak256, { sortPairs: true });
-
   // Return
-  const calculated = tree.getHexRoot();
+  let calculated: string;
+  try {
+    calculated = tree.root_js();
+  } finally {
+    tree.free(); // handle memory leaks
+  }
+
   return {
     root: calculated === "0x" ? HashZero : calculated,
     tree,
   };
 };
+
```

Comment:
I didn't notice this above because it's not part of the review, but this `root: calculated === "0x" ? HashZero : calculated,` can just be `root: calculated` because the lib already returns `HashZero`. With this simplification, you might as well also rename `calculated` to `root` and then have `return { root }`, which could just become `return root` after removing `tree`.